stack/Stack.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+class MainClass
+{
+    public static void Main()
+    {
+        Stack<string> test = new Stack<string>();
+
+        test.Push("123");
+        test.Push("test");
+        test.Push("rawr");
+        test.Push("Pancake");
+        test.Push("Yummy");
+        test.Push("321");
+        test.Push(":thinking:");
+
+        test.Swap();
+
+        test.Rotate(4, true);
+
+        test.Duplicate();
+
+        test.Peek();
+
+        while(!(test.IsEmpty()))
+        {
+            test.Pop();
+        }
+    }
+}
+
+class Stack<T>
+{
+    private LinkedList<T> stack;
+
+    /**
+    * Constructs an empty stack
+    */
+    public Stack()
+    {
+        stack = new LinkedList<T>();
+    }
+
+    /**
+    * Adds element at the top of the stack
+    */
+    public void Push(T data)
+    {
+        stack.AddLast(data);
+    }
+
+    /**
+    * Removes the element at the top of the stack
+    */
+    public void Pop()
+    {
+        if (IsEmpty())
+            return;
+
+        stack.RemoveLast();
+    }
+
+    /**
+    * Returns true if the stack is empty
+    */
+    public bool IsEmpty()
+    {
+        return (!stack.Any());
+    }
+
+    /**
+    * Swaps places between the last and second last element of the stack
+    */
+    public void Swap()
+    {
+        if (IsEmpty())
+            return;
+
+        T last = stack.Last();
+
+        T secondLast = stack.ElementAt(stack.Count() - 2);
+
+        Pop();
+        Pop();
+
+        Push(last);
+        Push(secondLast);
+    }
+
+    /**
+    * Returms the last element of the stack
+    */
+    public T Peek()
+    {
+        if (IsEmpty())
+            return default(T);
+
+        return stack.Last();
+    }
+
+    /**
+    * pops and pushes the same element twice
+    */
+    public void Duplicate()
```

Comment:
Same here. Do we need such a method? Is it implemented in any of the standard libraries?